### PR TITLE
Support Strings CEL library

### PR DESCRIPTION
**Pull request by pikachu0310**
Provide functionalities similar to Go's [strings](https://pkg.go.dev/strings) using the grpc.federation.strings library.
https://github.com/mercari/grpc-federation/issues/159

### CODE REVIEW

File: grpc/federation/cel/strings.go
Comment (shuheiktgw) [inline grpc/federation/cel/strings.go:95]: What is this comment? 🤔

File: grpc/federation/cel/strings.go
Comment (shuheiktgw) [inline grpc/federation/cel/strings.go:181]: Would you explain why we need this `OverloadFunc` even if we already have the index function? 🙏

**Review by shuheiktgw — COMMENTED** [review]
I left some questions but overall it looks good to me 👍

File: grpc/federation/cel/strings.go
Comment (pikachu0310) [reply grpc/federation/cel/strings.go:181]: I referred to the implementation linked below. While testing, I encountered an issue when trying to pass a Go `byte` type as an argument. It seems that trying to retrieve it using `cel.bytes` didn't work as expected. By defining the argument as `cel.string` instead, I was able to successfully pass and retrieve the value.

https://github.com/mercari/grpc-federation/blob/b948f38603ce28df91b0b67cd3f0af65b54f1ec4/grpc/federation/cel/time.go#L582-L612

File: grpc/federation/cel/strings.go
Comment (pikachu0310) [reply grpc/federation/cel/strings.go:95]: Sorry for the lack of explanation in the comments. I left a comment for the functions I determined to be impossible to implement. Except for `Cut`, the other comments are for functions that have arguments of types (like Go's `func` type) that cannot be received by CEL. As for `Cut`, I concluded it was impossible to implement because it needs to return two variables. 

Is there any way to return two variables in this case?

Comment (shuheiktgw) [reply grpc/federation/cel/strings.go:95]: We talked on this point directly and decided to update the comments to clarify why we cannot implement those functions.

File: grpc/federation/cel/strings.go
Comment (shuheiktgw) [reply grpc/federation/cel/strings.go:181]: Makes sense!

**Review by shuheiktgw — APPROVED** [review]
LGTM 👍 
Waiting for an approval from @utahta.

File: grpc/federation/cel/strings.go
Comment (utahta) [inline grpc/federation/cel/strings.go:20]: It seems that typeAdapter is not being used.  Is it possible to remove it?

Comment (pikachu0310) [reply grpc/federation/cel/strings.go]: You're right, it's not being used. I've removed it. Thank you for pointing it out.

File: grpc/federation/cel/strings.go
Comment (pikachu0310) [reply grpc/federation/cel/strings.go]: > Do we miss the Cut function?

Regarding "Cut," I found that it's feasible to implement, so I added the implementation.

**Review by utahta — APPROVED** [review]
LGTM!

**Review by shuheiktgw — APPROVED** [review]
🙆‍♂️

**Pull request merged**
